File: DecompiledGameScripts/cyberpunk/devices/lootContainer/lootContainerAccessPoint.swift

public class LootContainerAccessPoint extends AccessPoint {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as LootContainerAccessPointController;
  }

  public const func GetDevicePS() -> ref<LootContainerAccessPointControllerPS> {
    return this.GetController().GetPS();
  }

  private const func GetController() -> ref<LootContainerAccessPointController> {
    return this.m_controller as LootContainerAccessPointController;
  }

  public const func DeterminGameplayRole() -> EGameplayRole {
    return EGameplayRole.ControlOtherDevice;
  }
}

File: Definitions/Game_Definitions/classes/LootContainerAccessPoint.lua
---@meta
---@diagnostic disable

---@class LootContainerAccessPoint : AccessPoint
LootContainerAccessPoint = {}

---@return LootContainerAccessPoint
function LootContainerAccessPoint.new() return end

---@param props table
---@return LootContainerAccessPoint
function LootContainerAccessPoint.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function LootContainerAccessPoint:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function LootContainerAccessPoint:OnTakeControl(ri) return end

---@return EGameplayRole
function LootContainerAccessPoint:DeterminGameplayRole() return end

---@return LootContainerAccessPointController
function LootContainerAccessPoint:GetController() return end

---@return LootContainerAccessPointControllerPS
function LootContainerAccessPoint:GetDevicePS() return end

